src/gridfs-stream/upload.ts

```diff
@@ -208,8 +209,9 @@ export class GridFSBucketWriteStream extends Writable {
     this.state.streamEnd = true;
 
     if (callback) {
+      const cb = callback;
       this.once(GridFSBucketWriteStream.FINISH, (result: GridFSFile) => {
-        callback!(undefined, result);
+        cb(undefined, result);
```

Comment:
There is a good reason:
```typescript
let callbackish: Function | undefined = () => console.log('yay')
if(callbackish){
  doLater(() => {
    callbackish() // error potentially nullable
  });
}
callbackish = undefined;
```
Since doLater runs async, its possible callback could be made null again by the time it runs.
Updated the code with your suggestion

Reply:
right, but it's unlikely that it is reassigned. It would be nice if there was some sort of way to hint to the compiler that if a value was provided for `callback` then it would be immutable for the duration of the methods scope. :shrug: